overlord/snapstate/handlers.go

```diff
@@ -1365,6 +1485,22 @@ func (m *SnapManager) undoCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	// if any migration action was taken, try to undo it and rewrite state
+	if snapsup.MigratedToExposedHome {
+		if err := m.backend.RemoveExposedSnapHome(snapsup.InstanceName()); err != nil {
+			return err
+		}
+
+		snapsup.MigratedToExposedHome = false
+		snapst.MigratedToExposedHome = false
+
+		st.Lock()
+		err = writeMigrationStatus(t, snapst, snapsup)
```

Comment:
same asymmetry question, doCopyData doesn't manipulate the seq file or SnapState. Also again done potentially twice also releasing the lock twice